torch/csrc/jit/pickler.cpp

```diff
@@ -157,13 +162,37 @@ void Pickler::pushClass(PicklerClass cls) {
 }
 
 void Pickler::pushTensor(const IValue& ivalue) {
-  pushClass(PicklerClass::TENSOR);
+  if (tensor_table_ == nullptr) {
+    pushLiteralTensor(ivalue);
+  } else {
+    pushTensorReference(ivalue);
+  }
+}
+
+void Pickler::pushLiteralTensor(const IValue& ivalue) {
+  pushClass(PicklerClass::LITERAL_TENSOR);
+
+  auto tensor = ivalue.toTensor();
+  uint64_t record_size =
+      tensor.element_size() * tensor.storage().size();
+
+  IValue num_elements = tensor.sizes();
+  // TODO: making IValues does a useless copy of the storage
+  IValue storage_bytes =
+      std::string((char*)tensor.storage().data(), record_size);
+  IValue list = c10::ivalue::GenericList::create({storage_bytes, num_elements});
```

Comment:
When un-pickling only the last object on the stack gets popped off and passed to `__setstate__`, and to recreate the tensor we need both values hence the list wrapper. Creating `IValue`s from them lets us re-use the existing list serialization code instead of copying it here